src/components/modal/v2/parts/BodyContent.jsx

```diff
@@ -46,8 +46,15 @@ const BodyContent = () => {
     const content = useContent(viewName);
     const productMeta = useProductMeta(viewName);
 
-    const { headline, subheadline, qualifyingSubheadline = '', closeButtonLabel } = content;
+    const {
+        headline,
+        subheadline,
+        qualifyingSubheadline = '',
+        closeButtonLabel,
+        preapproval: { preapprovalHeadline, preapprovalSubHeadline }
+    } = content;
     const isQualifying = productMeta?.qualifying;
+    const isPreapproved = productMeta?.preapproved;
 
     const openProductList = () => setViewName('PRODUCT_LIST');
 
```

Comment:
```suggestion
    const {
        headline,
        subheadline,
        qualifyingSubheadline = '',
        closeButtonLabel,
    } = content;
    const isQualifying = productMeta?.qualifying;
    const isPreapproved = productMeta?.preapproved;
    const preapprovalHeadline = content?.preapproval?.preapprovalHeadline;
    const preapprovalSubHeadline = content?.preapproval?.preapprovalSubHeadline;
    
    const openProductList = () => setViewName('PRODUCT_LIST');

```
Since preapproval-related items can potentially be missing from the content (i.e. in the case of the regular upstream Pay in 4 modal), instead of restructuring we'll want to define those variables separately. That should prevent our non-checkout short term modals from erroring out.